src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -132,12 +132,30 @@ void SmithWatermanAlgorithm::launch(const SMatrix &_substitutionMatrix,
 }
 
 bool SmithWatermanAlgorithm::isValidParams() {
-    if (searchSeq.length() <= 0 || patternSeq.length() <= 0)
+    if (searchSeq.length() <= 0 || patternSeq.length() <= 0) {
         return false;
-    if (searchSeq.length() < patternSeq.length())
+    }
+    if (searchSeq.length() < patternSeq.length()) {
+        return false;
+    }
+    if (gapOpen >= 0 || gapExtension >= 0) {
         return false;
-    if (gapOpen >= 0 || gapExtension >= 0)
+    }
+    unsigned int memory = 0;
+    quint64 quint_memory = 0;
+    if (resultView == SmithWatermanSettings::MULTIPLE_ALIGNMENT) {
+        int pat_n = patternSeq.length();
+        unsigned int dirn = (4 + pat_n + 3) >> 2;
+        memory = pat_n * 2 * sizeof(int) + pat_n * 0x80 + matrixLength * dirn;
+        quint_memory = (quint64)pat_n * 2 * sizeof(int) + (quint64)pat_n * 0x80 + (quint64)matrixLength * (quint64)dirn;
+    } else if (resultView == SmithWatermanSettings::ANNOTATIONS) {
+        int pat_n = patternSeq.length();
+        memory = pat_n * 3 * sizeof(int) + pat_n * 0x80;
+        quint_memory = (quint64)pat_n * 3 * sizeof(int) + (quint64)pat_n * (quint64)0x80;
+    }
+    if (quint_memory != memory) {
```

Comment:
This check must be done in the same function with a malloc.
Repeating a complex computation in a separate place is incorrect: what if malloc code will change (will have +1 in the memory allocation size).

All other changes in parameters validation code are not related to the original issue: I would suggest removing them (keeping the original code) or filing another bug with a new problem description.